src/zeek-setup.cc

```diff
@@ -843,7 +856,7 @@ SetupResult setup(int argc, char** argv, Options* zopts)
 
 		if ( reporter->Errors() > 0 )
 			{
-			delete dns_mgr;
+			early_shutdown();
```

Comment:
Should `early_shutdown` also delete `dns_mgr`, since it looks like that deletion was lost in a few places?

Reply:
I've removed the manual deletes because ASAN otherwise complains (i.e., crashes the application). There's also this comment in line 426 about cleanup dependencies:

```
// broker_mgr, timer_mgr, supervisor, and dns_mgr are deleted via iosource_mgr
```

🙂